visualization/app/codeCharta/state/store/appSettings/cameraTarget/cameraTarget.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Vector3 } from "three"
+import { StoreService, StoreSubscriber } from "../../../store.service"
+import { IRootScopeService } from "angular"
+import { CameraTargetActions } from "./cameraTarget.actions"
+import { isActionOfType } from "../../../../util/reduxHelper"
+
+export interface CameraTargetSubscriber {
+	onStoreCameraTargetChanged(cameraTarget: Vector3)
+}
+
+export class CameraTargetService implements StoreSubscriber {
+	private static CAMERA_TARGET_CHANGED_EVENT = "camera-target-changed"
```

Comment:
JavaScript does support private properties as well (Using the #hash). It would be good to use that instead of TypeScript private. That way there's more interoperability and there's less to transpile.

I do not know what Node.js versions this should support though. It's only available in Node.js => v12.x.